cartridge/roles.lua

```diff
@@ -183,22 +183,23 @@ local function get_all_roles()
     return ret
 end
 
---- List registered roles names.
+--- List all registered roles.
 --
--- Hidden roles are not listed as well as permanent ones.
+-- Listed: name and flags: hidden, implies_router, implies_storage.
 --
 -- @function get_known_roles
 -- @local
--- @treturn {string,..}
+-- @treturn {{name = string, hidden = boolean, implies_router = boolean, implies_storage = boolean}, ...}
 local function get_known_roles()
     local ret = {}
 
     for _, role in ipairs(vars.roles_by_number) do
-        if not role.M.permanent
-        and not role.M.hidden
-        then
-            table.insert(ret, role.role_name)
-        end
+        table.insert(ret, {
+            name = role.role_name,
+            hidden = role.M.permanent or role.M.hidden or false,
+            implies_router = role.M.implies_router or false,
+            implies_storage = role.M.implies_storage or false,
+        })
```

Comment:
This change is incompatible.

For sure it'll break [cartridge-cli tests](https://github.com/tarantool/cartridge-cli/blob/master/test/utils.py#L865). They don't expect to see hidden roles in the list.

Changing the return format is a bad idea too. We can't be sure nobody uses it.

Reply:
As a result, I have added options `implies_router` and `implies_storage` in GraphQL, but not to `get_known_roles`. It should not break tests in cartridge-cli and any user's code.